test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/DotnetCommand.cs

```diff
@@ -15,6 +17,41 @@ public DotnetCommand()
         public DotnetCommand(string dotnetUnderTest)
             : base(dotnetUnderTest)
         {
+            // Some scripts (e.g. RunCsc/RunVbc) depend on the correct `dotnet`
+            // being in the PATH. When tests use `DotnetUnderTest.WithBackwardsCompatibleRuntimes`,
+            // this means that the up-to-date dotnet command is resolved by the
+            // old-runtime-targeting RunCsc/Roslyn, causing compatability issues.
+            // So, detect if we're running a dotnet that's not just a plain "dotnet",
+            // and if so, make sure it's on the PATH.
+            var containingDirectory = Path.GetDirectoryName(dotnetUnderTest);
+            if (!string.IsNullOrEmpty(containingDirectory))
+            {
+                // if it's already set, use that
+                string path;
+                if (!Environment.TryGetValue("PATH", out path))
+                {
+                    // otherwise grab it from the environment
```

Comment:
I believe that the comments are necessary for understanding this code, as they explain the why, instead of just the how. I don't think a method name is adequate for describing the subtleties going on in this code. And for the feedback that understanding comments is required before changes to the code can be made: Yes, that is correct, and very important, because any changes done without a strong understanding of what this code is doing may introduce subtle bugs.

Additionally, I've just pushed a commit that takes into account feedback yesterday, to use `DOTNET_HOST_PATH` instead of modifying `PATH` here. This cuts out some of the comments - could you let me know if you still think there are too many comments, and if so, what should be done to resolve them?